dvc/repo/checkout.py

```diff
@@ -44,15 +37,14 @@ def checkout(self, target=None, with_deps=False, force=False, recursive=False):
 
     with self.state:
         _cleanup_unused_links(self, all_stages)
-        progress_callback = get_progress_callback(stages)
-
-        for stage in stages:
-            if stage.locked:
-                logger.warning(
-                    "DVC-file '{path}' is locked. Its dependencies are"
-                    " not going to be checked out.".format(path=stage.relpath)
-                )
-
-            stage.checkout(force=force, progress_callback=progress_callback)
-        if progress_callback:
-            progress_callback.finish("Checkout finished!")
+        with Tqdm(total=get_all_files_numbers(stages), unit="file") as pbar:
+            for stage in stages:
+                if stage.locked:
+                    logger.warning(
+                        "DVC-file '{path}' is locked. Its dependencies are"
+                        " not going to be checked out.".format(
+                            path=stage.relpath
+                        )
+                    )
+
+                stage.checkout(force=force, progress_callback=pbar.update_desc)
```

Comment:
@efiop the blank end bar bug seems to be because this `progress_callback` is never called. Replacing `progress_callback=pbar.update_desc` with `progress_callback=print` prints nothing. I can't work out why.

Reply:
@casperdcl Ah, that is a good point. So the reason is that in my reproducer script, I'm not removing the files in workspace(those are "links" from cache), so `checkout`, which is part of the `pull`, has nothing left to do. The shortcurcuitting is happening in https://github.com/iterative/dvc/blob/0.55.0/dvc/remote/base.py#L766 with another one happening in https://github.com/iterative/dvc/blob/0.55.0/dvc/remote/base.py#L772 , so looks like we need to call progress_callback there. This has been a bug, which we didn't notice because of the final `finish()` that we've had before. Great catch!

Reply:
oh this would have been impossible for me to find since I haven't merged master into my branch yet, so `progress_callback` was unconditionally called. Hah.

Reply:
ah I was looking at `dvc.remote.base._checkout` rather than `dvc.remote.base.checkout`